### PR TITLE
Adds eye color overrides, converts cult/weed/silverscale eye effects to use them

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Eye color effects are now controlled via a new overrides system, which should prevent conflicts between different sources - such as smoking weed, becoming a cultist, stopping your weed and getting normal eyes (and subsequently getting your red eyes back when you get deconverted)

## Why It's Good For The Game

Should ensure that different eye color effects don't collide between each other. Also need this for a future feature I wanna add.

## Changelog
:cl:
refactor: Eye color effects are now controlled via a new overrides system, which should prevent conflicts between different sources.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
#45097

**Comment by SmArtKar** [comment]
> #45097

Unrelated to what i'm doing here actually

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1029,6 +1029,50 @@
...
+
+/mob/living/carbon/human/proc/get_right_eye_color()
+	if (!LAZYLEN(eye_color_right_overrides))
+		return eye_color_right
+
+	var/eye_color = eye_color_right
+	var/priority
+	for (var/override_priority in eye_color_right_overrides)
+		if (text2num(override_priority) > priority)
+			priority = text2num(override_priority)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/human.dm:1060]:
```suggestion
		var/new_priority = text2num(override_priority)
		if (new_priority > priority)
			priority = new_priority
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1029,6 +1029,50 @@
...
+
+/mob/living/carbon/human/proc/get_left_eye_color()
+	if (!LAZYLEN(eye_color_left_overrides))
+		return eye_color_left
+
+	var/eye_color = eye_color_left
+	var/priority
+	for (var/override_priority in eye_color_left_overrides)
+		if (text2num(override_priority) > priority)
+			priority = text2num(override_priority)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/human.dm:1072]:
```suggestion
		var/new_priority = text2num(override_priority)
		if (new_priority > priority)
			priority = new_priority
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -1029,6 +1029,50 @@
...
+	if (update_body)
+		update_body()
+
+/mob/living/carbon/human/proc/add_eye_color_right(color, color_priority, update_body = TRUE)
+	LAZYSET(eye_color_right_overrides, "[color_priority]", color)
+	if (update_body)
+		update_body()
+
+/mob/living/carbon/human/proc/add_eye_color(color, color_priority, update_body = TRUE)
+	add_eye_color_left(color, color_priority, update_body = FALSE)
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/human.dm:1043]:
```suggestion
	add_eye_color_left(color, color_priority, update_body = update_body)
```

**Review by Ghommie — APPROVED** [review]

**Review by Time-Green — CHANGES_REQUESTED** [review]
sad it's not on the eyes itself but otherwise okay

File: code/modules/mob/living/carbon/human/human.dm (hunk 3 above)
Comment (SmArtKar) [reply code/modules/mob/living/carbon/human/human.dm:1043]: We call update_body on the second one if its set to TRUE, no need to call it twice

**Pull request merged**